packages/experiments/src/components/Text/Text.tsx

```diff
@@ -1,173 +1,13 @@
-
-  let themeType;
-
-  const paletteColor = color as keyof IPalette;
-  const semanticColor = color as keyof ISemanticColors;
-
-  if (fontStyle && fonts[fontStyle]) {
-    // if a general style is specified, use that for the theme
-    themeType = fonts[fontStyle];
-  } else {
-    if (type && typography.types[type]) {
-      // if a general type is specified, use it to initialize the theme
-      themeType = typography.types[type];
-    }
-
-    // next, look for individual properties and use them if they are present
-    const fontFamily =
-      family && typography.families[family] // if family prop is passed in and exists in typography.families, we'll use it
-        ? typography.families[family]
-        : typography.families.default; // otherwise use default
-
-    const fontWeight = weight && typography.weights[weight] ? typography.weights[weight] : typography.weights.default;
-
-    const fontSize = size && typography.sizes[size] ? typography.sizes[size] : typography.sizes.medium;
-
-    if (themeType) {
-      // if a type was specified and individual properties were also specified, use individual properties
-      if (family && typography.families[family]) {
-        themeType.fontFamily = fontFamily;
-      }
-      if (weight && typography.weights[weight]) {
-        themeType.fontWeight = fontWeight;
-      }
-      if (size && typography.sizes[size]) {
-        themeType.fontSize = fontSize;
-      }
-    } else {
-      themeType = {
-        fontFamily: fontFamily,
-        fontWeight: fontWeight,
-        fontSize: fontSize
-      };
-    }
-  }
-
-  // use fonts to set these properties
-  const mozOsxFontSmoothing =
-    fontStyle && fonts[fontStyle] ? fonts[fontStyle].MozOsxFontSmoothing : fonts.medium.MozOsxFontSmoothing;
-
-  const webkitFontSmoothing =
-    fontStyle && fonts[fontStyle] ? fonts[fontStyle].WebkitFontSmoothing : fonts.medium.WebkitFontSmoothing;
-
-  const themeProperties = {
-    mozOsxFontSmoothing: mozOsxFontSmoothing,
-    webkitFontSmoothing: webkitFontSmoothing,
-    color:
-      semanticColor && semanticColors[semanticColor] // use semanticColors to set color if it exists
-        ? semanticColors[semanticColor]
-        : paletteColor && palette[paletteColor]
-          ? palette[paletteColor]
-          : palette.neutralPrimary // otherwise, use palette
-  };
-
-  return {
-    root: [
-      themeType,
-      themeProperties,
-      {
-        display: inline ? 'inline' : 'block'
-      },
-      !wrap && {
-        whiteSpace: 'nowrap',
-        overflow: 'hidden',
-        textOverflow: 'ellipsis'
-      },
-      props.className
-    ]
-  } as ITextStyles;
-};
+import { createComponent } from '../../Foundation';
+import { ITextProps, ITextStyles } from './Text.types';
+import { TextView as view } from './Text.view';
+import { TextStyles as styles } from './Text.styles';
 
 export const Text: React.StatelessComponent<ITextProps> = createComponent<ITextProps, ITextStyles>({
```

Comment:
This file should be a NEW file called Text.ts, so everything previously here is recognized as a rename to Text.base.ts and preserves history (presently, you will show up on the entire file's git blame and history as well)